pkg/daemon/daemon.go

```diff
@@ -333,18 +382,42 @@ func (d *daemon) AddPeriodicUpdate() {
 	log.Info().Msg("add periodic update finished")
 }
 
+// get GUID from Pod's network
+func getPodGUIDForNetwork(pod *kapi.Pod, networkName string) (net.HardwareAddr, error) {
+	networks, netErr := netAttUtils.ParsePodNetworkAnnotation(pod)
+	if netErr != nil {
+		return nil, fmt.Errorf("failed to read pod networkName annotations pod namespace %s name %s, with error: %v",
+			pod.Namespace, pod.Name, netErr)
+	}
+
+	network, netErr := utils.GetPodNetwork(networks, networkName)
+	if netErr != nil {
+		return nil, fmt.Errorf("failed to get pod networkName spec %s with error: %v", networkName, netErr)
+	}
+
+	if !utils.IsPodNetworkConfiguredWithInfiniBand(network) {
+		return nil, errors.Errorf(errors.NotValidError, "network %+v is not InfiniBand configured", network)
```

Comment:
So this will now emit an error log instead of a warning as before.
i think we are OK with that